msalautomationapp/src/androidTest/java/com/microsoft/identity/client/msal/automationapp/testpass/broker/TestCase833513.java

```diff
@@ -0,0 +1,140 @@
+import com.microsoft.identity.client.ui.automation.broker.BrokerMicrosoftAuthenticator;
+import com.microsoft.identity.client.ui.automation.broker.ITestBroker;
+import com.microsoft.identity.client.ui.automation.interaction.PromptHandlerParameters;
+import com.microsoft.identity.client.ui.automation.interaction.PromptParameter;
+import com.microsoft.identity.client.ui.automation.interaction.microsoftsts.AdfsPromptHandler;
+import com.microsoft.identity.client.ui.automation.utils.UiAutomatorUtils;
+import com.microsoft.identity.internal.testutils.labutils.LabConfig;
+import com.microsoft.identity.internal.testutils.labutils.LabConstants;
+import com.microsoft.identity.internal.testutils.labutils.LabUserHelper;
+import com.microsoft.identity.internal.testutils.labutils.LabUserQuery;
+
+import org.junit.Assert;
+import org.junit.Test;
+
+// End My Shift - In Shared device mode, only account from the same tenant should be able to acquire token.
+// https://identitydivision.visualstudio.com/DevEx/_workitems/edit/833513
+public class TestCase833513 extends AbstractMsalBrokerTest {
+
+    @Test
+    public void test_833513() throws MsalException, InterruptedException {
+        // pca should be in MULTIPLE account mode starting out
+        Assert.assertTrue(mApplication instanceof MultipleAccountPublicClientApplication);
+
+        //we should NOT be in shared device mode
+        Assert.assertFalse(mApplication.isSharedDevice());
+
+        // perform shared device registration
+        mBroker.performSharedDeviceRegistration(
+                mLoginHint, LabConfig.getCurrentLabConfig().getLabUserPassword()
+        );
+
+        // re-create PCA after device registration
+        mApplication = PublicClientApplication.create(mContext, getConfigFileResourceId());
+
+        // pca should now be in SINGLE account mode
+        Assert.assertTrue(mApplication instanceof SingleAccountPublicClientApplication);
+
+        // we should be in shared device mode
+        Assert.assertTrue(mApplication.isSharedDevice());
+
+        // query to load a user from a different tenant that was used for WPJ
+        final LabUserQuery query = new LabUserQuery();
+        query.userType = LabConstants.UserType.FEDERATED;
+        query.federationProvider = LabConstants.FederationProvider.ADFS_V3;
+
+        final String username = LabUserHelper.loadUserForTest(query);
+        String password = LabConfig.getCurrentLabConfig().getLabUserPassword();
+
+        final SingleAccountPublicClientApplication singleAccountPCA =
+                (SingleAccountPublicClientApplication) mApplication;
+
+        // try sign in with an account from a different tenant
+        // passing null for latch as we don't need to receive the result from this call
+        // we just want to get into the webview and look for the error in AAD page
+        singleAccountPCA.signIn(mActivity, username, mScopes, successfulInteractiveCallback(null));
```

Comment:
It's not matching the test case (the test case asks the tester to launch AzureSample). 

Let's make sure they're matching. In this case, I'm fine either making changes here or there.

Reply:
I'll update the test case in ADO.....as I remember that I did it this way because I realized we didn't need to go into AzureSampleApp to test this

Reply:
Updated test case in ADO: https://identitydivision.visualstudio.com/DevEx/_workitems/edit/833513